### PR TITLE
EDAC: IBECC: Refactor tests for new API

### PR DESCRIPTION
**Pull request by finikorg**
Refactor tests for new ZTEST:

- Use new ZTEST API right way
- Using new helpers like `Z_TEST_SKIP_IFNDEF`. 
- Make also all tests independent, suitable for shuffle.

### CODE REVIEW

**Review by yperess — DISMISSED** [review]

**Comment by yperess** [comment]
@finikorg @nashif  I'm seeing a lot of calls to TC_PRINT in this test, my understanding was that TC_ macros are meant to be internal to the test subsystem to be able and keep an easy format, should these be logs instead?

**Comment by finikorg** [comment]
> @finikorg @nashif I'm seeing a lot of calls to TC_PRINT in this test, my understanding was that TC_ macros are meant to be internal to the test subsystem to be able and keep an easy format, should these be logs instead?

I added commit [af9f5da](https://github.com/zephyrproject-rtos/zephyr/pull/48170/commits/af9f5da28439d018f413e31b127adf183488a37d) replacing TC_PRINT to logging api. For the coverage test I source C file so I cannot change the same way.

**Review by yperess — DISMISSED** [review]

File: tests/subsys/edac/ibecc/src/ibecc.c [hunk 1]
```diff
@@ -38,6 +38,9 @@ K_APP_BMEM(default_part) static volatile uint32_t error_type;
 K_APP_BMEM(default_part) static volatile uint64_t error_address;
 K_APP_BMEM(default_part) static volatile uint16_t error_syndrome;
 
+/* Keep track or correctable and uncorrectable errors */
+unsigned int errors_correctable, errors_uncorrectable;
```
Comment (jhedberg) [inline tests/subsys/edac/ibecc/src/ibecc.c:42]: static perhaps?

Comment (finikorg) [reply tests/subsys/edac/ibecc/src/ibecc.c]: Fixed

**Review by jhedberg — DISMISSED** [review]

**Review by enjiamai — APPROVED** [review]

**Comment by finikorg** [comment]
@nashif could you review this PR

**Review by jhedberg — APPROVED** [review]

**Pull request merged**